app/controllers/sns_controller.rb

```diff
@@ -0,0 +1,53 @@
+# These endpoints allow the Amazon SES service to notify us when there are bounced
+# emails or complaints (ex we are marked as spam). The expected JSON is particular to
+# Amazon's SNS notification service, and thus will not work for other services.
+class SnsController < ApplicationController
+  protect_from_forgery with: :null_session
+  before_filter :verify_amazon_authorize_key
+
+  def bounce
+    begin
+      success = true
+      message = JSON.parse(request.body.read)['Message']
+      bounce_arr = JSON.parse(message)['bounce']['bouncedRecipients']
+      bounce_arr.each do |recipient|
+        Bounce.create(email: recipient['emailAddress'].downcase)
+      end
+    rescue
```

Comment:
Yeah, I based this on the old code but that seems better. It was hard to find errors when they kept getting rescued.